projects/igniteui-angular/src/lib/core/styles/spec/_mixins.spec.scss

```diff
@@ -43,4 +43,236 @@
+        @include assert() {
+            @include output() {
+                @include igx-palette-vars($my-palette);
+            }
+
+            @include contains() {
+                --igx-primary-50: #ffc0c0;
+                --igx-primary-50-contrast: black;
+                --igx-primary-100: #faa7a7;
+                --igx-primary-100-contrast: black;
+                --igx-primary-200: #f16d6d;
+                --igx-primary-200-contrast: black;
+                --igx-primary-300: #f23c3c;
+                --igx-primary-300-contrast: black;
+                --igx-primary-400: #f71d1d;
+                --igx-primary-400-contrast: black;
+                --igx-primary-500: red;
+                --igx-primary-500-contrast: black;
+                --igx-primary-600: #e50000;
+                --igx-primary-600-contrast: white;
+                --igx-primary-700: #d20000;
+                --igx-primary-700-contrast: white;
+                --igx-primary-800: #be0000;
+                --igx-primary-800-contrast: white;
+                --igx-primary-900: #a80000;
+                --igx-primary-900-contrast: white;
+                --igx-primary-A100: #ff5555;
+                --igx-primary-A100-contrast: black;
+                --igx-primary-A200: #ff2727;
+                --igx-primary-A200-contrast: black;
+                --igx-primary-A400: #eb0000;
+                --igx-primary-A400-contrast: white;
+                --igx-primary-A700: #aa0000;
+                --igx-primary-A700-contrast: white;
+                --igx-secondary-50: #41ff41;
+                --igx-secondary-50-contrast: black;
+                --igx-secondary-100: #2ff42f;
+                --igx-secondary-100-contrast: black;
+                --igx-secondary-200: #13cc13;
+                --igx-secondary-200-contrast: black;
+                --igx-secondary-300: #0ba30b;
+                --igx-secondary-300-contrast: black;
+                --igx-secondary-400: #059005;
+                --igx-secondary-400-contrast: black;
+                --igx-secondary-500: green;
+                --igx-secondary-500-contrast: white;
+                --igx-secondary-600: #006600;
+                --igx-secondary-600-contrast: white;
+                --igx-secondary-700: #005300;
+                --igx-secondary-700-contrast: white;
+                --igx-secondary-800: #003f00;
+                --igx-secondary-800-contrast: white;
+                --igx-secondary-900: #002900;
+                --igx-secondary-900-contrast: white;
+                --igx-secondary-A100: #00d500;
+                --igx-secondary-A100-contrast: black;
+                --igx-secondary-A200: #00a700;
+                --igx-secondary-A200-contrast: black;
+                --igx-secondary-A400: #006c00;
+                --igx-secondary-A400-contrast: white;
+                --igx-secondary-A700: #002b00;
+                --igx-secondary-A700-contrast: white;
+            }
```

Comment:
This is a bit too much for my taste. What this test should check, instead, is whether the keys contained in the $palette map are part of the content. You can do that using a for loop inside the contains block.